src/ansys/fluent/core/services/field_data.py

```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
+
+
+class Vector:
+    """Stores the data in form of a vector ``(x, y, z)``."""
+
+    def __init__(self, x, y, z):
+        self._x = x
+        self._y = y
+        self._z = z
+
+    @property
+    def x(self) -> float:
+        return self._x
+
+    @property
+    def y(self) -> float:
+        return self._y
+
+    @property
+    def z(self) -> float:
+        return self._z
+
+
+def _resolve_into_array_of_vectors(data):
+    if data.size % 3:
+        raise ValueError(
+            "Dataset must be resolved as a set of vectors."
+            "The length of the dataset should always be in multiples of 3."
+        )
+    data.shape = data.size // 3, 3
+
+
+class VectorFieldData(BaseFieldData):
+    """Provides a container for vector field data."""
+
+    class VectorData(Vector):
+        """Stores and provides the data as a vector."""
+
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data, scale):
+        _resolve_into_array_of_vectors(data)
+        self._scale = scale
+        super().__init__(i_d, [VectorFieldData.VectorData(x, y, z) for x, y, z in data])
+
+    @property
+    def scale(self) -> float:
+        return self._scale
+
+
+class Vertices(BaseFieldData):
+    """Provides a container for the vertex data."""
+
+    class Vertex(Vector):
+        """Stores and provides the data as a vector of a vertex."""
+
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data):
+        _resolve_into_array_of_vectors(data)
+        super().__init__(i_d, [(Vertices.Vertex(x, y, z)) for x, y, z in data])
+
+
+class FacesCentroid(BaseFieldData):
+    """Provides the container for the face centroid data."""
+
+    class Centroid(Vector):
+        """Stores and provides the face centroid data as a vector."""
+
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data):
+        _resolve_into_array_of_vectors(data)
+        super().__init__(i_d, [(FacesCentroid.Centroid(x, y, z)) for x, y, z in data])
+
+
+class FacesConnectivity(BaseFieldData):
+    """Provides the container for the face connectivity data."""
+
+    class Faces:
+        """Stores and provides the face connectivity data as an array."""
+
+        def __init__(self, node_count, node_data):
+            self.node_count = node_count
+            self.node_data = node_data
+
+    def __init__(self, i_d, data):
+        faces_data = []
+        i = 0
+
+        while i < len(data):
+            end = i + 1 + data[i]
+            faces_data.append(FacesConnectivity.Faces(data[i], data[i + 1 : end]))
+            i = end
+
+        super().__init__(i_d, faces_data)
+
+
+class FacesNormal(BaseFieldData):
+    """Provides the container for the face normal data."""
+
+    class Normal(Vector):
+        """Stores and provides the face's normal data as a vector."""
```

Comment:
I'm not sure about my earlier suggestion for ussing the possessive here. Based on the description for the parent class, should we be saving "face normal data" in the description for the ``Normal`` class--or should we be using the possessive  "face's normal data`` in the parent ``FacesNormal`` class? It seems the terminology should match.

Reply:
Actually faces means a number of faces (surfaces). That's why the terminology FacesNormal and that's why face's normal does not make much sense in this case. You have any suggestions on what can be done to maintain consistency? @PipKat

Reply:
"face normal data" makes sense here.

Reply:
@prmukherj I'm fine with face normal data. However, the use of pathlines (pluarl) is fine. Let me find the line.